webapp/client/src/components/BackLink.test.js

```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import BackLink from "./BackLink";
+
+it("should render the link text", () => {
+  render(<BackLink text="Back" url="/some/path" />);
+  expect(screen.getByText("Back")).toBeInTheDocument();
+});
+
+it("should link to the URL", () => {
+  render(<BackLink text="Back" url="/some/path" />);
+  expect(screen.getByText("Back").closest("a")).toHaveAttribute(
+    "href",
+    expect.stringContaining("/some/path")
```

Comment:
I wouldn't call this nesting, although it has multiple ways of failing:
- `screen.getByText("Back").closest("a")` tries to find the specified element (and fails if it can't find it)
- `expect(screen.getByText("Back").closest("a"))` just wraps that element so we can call matchers like `toHaveAttribute` on it
- `expect.stringContaining(...)` is a custom matcher that tells the test framework what the value of the `href` attribute should look like

Does that make sense? Maybe I misunderstood your question…

More here: https://testing-library.com/docs/react-testing-library/intro/

Reply:
No no, you got my question correctly. I just find the syntax unintuitive. But it works as I thought, so thanks. :)